### ISSUE TITLE
Restore default flag -Wmaybe-uninitialized

### ISSUE THREAD

**Issue by fredrik-johansson** [flintlib/flint#2224]
We turned this off because it was giving too many false positives, but it's one of the warnings that is actually useful when it works.

Currently with my gcc I get the following warnings. Maybe we can find workarounds or selectively turn off the warning just in those locations.

```
/home/fredrik/src/flint/src/nmod_poly/multi_crt.c: In function ‘nmod_poly_multi_crt_precompute’:
/home/fredrik/src/flint/src/nmod_poly/multi_crt.c:288:15: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
  288 |     success = nmod_poly_multi_crt_precompute_p(P, m, len);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/fredrik/src/flint/src/nmod_poly/multi_crt.c:194:5: note: by argument 2 of type ‘const nmod_poly_struct * const*’ to ‘nmod_poly_multi_crt_precompute_p’ declared here
  194 | int nmod_poly_multi_crt_precompute_p(
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



/home/fredrik/src/flint/src/fmpz_poly/evaluate_horner_d_2exp.c: In function ‘_fmpz_poly_evaluate_horner_d_2exp2’:
/home/fredrik/src/flint/src/fmpz_poly/evaluate_horner_d_2exp.c:167:9: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
  167 |     v = _fmpz_poly_evaluate_horner_d_2exp2_precomp(exp, p, p_exp, n, d, dexp);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/fredrik/src/flint/src/fmpz_poly/evaluate_horner_d_2exp.c:108:8: note: by argument 2 of type ‘const double *’ to ‘_fmpz_poly_evaluate_horner_d_2exp2_precomp’ declared here
  108 | double _fmpz_poly_evaluate_horner_d_2exp2_precomp(slong * exp, const double * poly, const slong * poly_exp, slong n, double d, slong dexp)
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/fredrik/src/flint/src/fmpz_poly/evaluate_horner_d_2exp.c:167:9: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
  167 |     v = _fmpz_poly_evaluate_horner_d_2exp2_precomp(exp, p, p_exp, n, d, dexp);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/fredrik/src/flint/src/fmpz_poly/evaluate_horner_d_2exp.c:108:8: note: by argument 3 of type ‘const slong *’ {aka ‘const long int *’} to ‘_fmpz_poly_evaluate_horner_d_2exp2_precomp’ declared here
  108 | double _fmpz_poly_evaluate_horner_d_2exp2_precomp(slong * exp, const double * poly, const slong * poly_exp, slong n, double d, slong dexp)
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



/home/fredrik/src/flint/src/mpoly/min_fields.c: In function ‘mpoly_min_fields_ui_sp’:
/home/fredrik/src/flint/src/mpoly/min_fields.c:42:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
   42 |     mpoly_unpack_vec_ui(min_fields, pmin, bits, mctx->nfields, 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/fredrik/src/flint/src/mpoly/min_fields.c:12:
/home/fredrik/src/flint/src/mpoly.h:899:6: note: by argument 2 of type ‘const ulong *’ {aka ‘const long unsigned int *’} to ‘mpoly_unpack_vec_ui’ declared here
  899 | void mpoly_unpack_vec_ui(ulong * exp1, const ulong * exp2, flint_bitcnt_t bits, slong nfields, slong len);
      |      ^~~~~~~~~~~~~~~~~~~
/home/fredrik/src/flint/src/mpoly/min_fields.c: In function ‘mpoly_min_fields_fmpz’:
/home/fredrik/src/flint/src/mpoly/min_fields.c:79:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
   79 |     mpoly_unpack_vec_fmpz(min_fields, pmin, bits, mctx->nfields, 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/fredrik/src/flint/src/mpoly/min_fields.c:12:
/home/fredrik/src/flint/src/mpoly.h:901:6: note: by argument 2 of type ‘const ulong *’ {aka ‘const long unsigned int *’} to ‘mpoly_unpack_vec_fmpz’ declared here
  901 | void mpoly_unpack_vec_fmpz(fmpz * exp1, const ulong * exp2, flint_bitcnt_t bits, slong nfields, slong len);
      |      ^~~~~~~~~~~~~~~~~~~~~



/home/fredrik/src/flint/src/fmpz_mod_mpoly/set_coeff_fmpz_fmpz.c: In function ‘fmpz_mod_mpoly_set_coeff_fmpz_fmpz’:
/home/fredrik/src/flint/src/fmpz_mod_mpoly/set_coeff_fmpz_fmpz.c:104:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
  104 |     _fmpz_mod_mpoly_set_coeff_fmpz_fmpz(A, c, newexp, ctx);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/fredrik/src/flint/src/fmpz_mod_mpoly/set_coeff_fmpz_fmpz.c:16:6: note: by argument 3 of type ‘const fmpz *’ {aka ‘const long int *’} to ‘_fmpz_mod_mpoly_set_coeff_fmpz_fmpz’ declared here
   16 | void _fmpz_mod_mpoly_set_coeff_fmpz_fmpz(
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



/home/fredrik/src/flint/src/arb/dot_fmpz.c: In function ‘arb_dot_fmpz’:
/home/fredrik/src/flint/src/arb/dot_fmpz.c:141:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
  141 |     arb_dot(res, initial, subtract, x, xstep, t, 1, len, prec);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/fredrik/src/flint/src/arb/dot_fmpz.c:12:
/home/fredrik/src/flint/src/arb.h:416:6: note: by argument 6 of type ‘arb_srcptr’ {aka ‘const arb_struct *’} to ‘arb_dot’ declared here
  416 | void arb_dot(arb_t res, const arb_t initial, int subtract,
      |      ^~~~~~~



/home/fredrik/src/flint/src/arb/dot_siui.c: In function ‘arb_dot_siui’:
/home/fredrik/src/flint/src/arb/dot_siui.c:109:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
  109 |     arb_dot(res, initial, subtract, x, xstep, t, 1, len, prec);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/fredrik/src/flint/src/arb/dot_siui.c:12:
/home/fredrik/src/flint/src/arb.h:416:6: note: by argument 6 of type ‘arb_srcptr’ {aka ‘const arb_struct *’} to ‘arb_dot’ declared here
  416 | void arb_dot(arb_t res, const arb_t initial, int subtract,
      |      ^~~~~~~




/home/fredrik/src/flint/src/arb/dot_uiui.c: In function ‘arb_dot_uiui’:
/home/fredrik/src/flint/src/arb/dot_uiui.c:100:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
  100 |     arb_dot(res, initial, subtract, x, xstep, t, 1, len, prec);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/fredrik/src/flint/src/arb/dot_uiui.c:12:
/home/fredrik/src/flint/src/arb.h:416:6: note: by argument 6 of type ‘arb_srcptr’ {aka ‘const arb_struct *’} to ‘arb_dot’ declared here
  416 | void arb_dot(arb_t res, const arb_t initial, int subtract,
      |      ^~~~~~~


/home/fredrik/src/flint/src/acb/dot_fmpz.c: In function ‘acb_dot_fmpz’:
/home/fredrik/src/flint/src/acb/dot_fmpz.c:141:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
  141 |     arb_dot(((arb_ptr) res) + 0, (initial == NULL) ? NULL : ((arb_srcptr) initial) + 0, subtract, ((arb_srcptr) x) + 0, 2 * xstep, t, 1, len, prec);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/fredrik/src/flint/src/acb.h:21,
                 from /home/fredrik/src/flint/src/acb/dot_fmpz.c:12:
/home/fredrik/src/flint/src/arb.h:416:6: note: by argument 6 of type ‘arb_srcptr’ {aka ‘const arb_struct *’} to ‘arb_dot’ declared here
  416 | void arb_dot(arb_t res, const arb_t initial, int subtract,


/home/fredrik/src/flint/src/acb/dot_si.c: In function ‘acb_dot_si’:
/home/fredrik/src/flint/src/acb/dot_si.c:72:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
   72 |     arb_dot(((arb_ptr) res) + 0, (initial == NULL) ? NULL : ((arb_srcptr) initial) + 0, subtract, ((arb_srcptr) x) + 0, 2 * xstep, t, 1, len, prec);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/fredrik/src/flint/src/acb.h:21,
                 from /home/fredrik/src/flint/src/acb/dot_si.c:12:
/home/fredrik/src/flint/src/arb.h:416:6: note: by argument 6 of type ‘arb_srcptr’ {aka ‘const arb_struct *’} to ‘arb_dot’ declared here
  416 | void arb_dot(arb_t res, const arb_t initial, int subtract,


/home/fredrik/src/flint/src/acb/dot_siui.c: In function ‘acb_dot_siui’:
/home/fredrik/src/flint/src/acb/dot_siui.c:110:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
  110 |     arb_dot(((arb_ptr) res) + 0, (initial == NULL) ? NULL : ((arb_srcptr) initial) + 0, subtract, ((arb_srcptr) x) + 0, 2 * xstep, t, 1, len, prec);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/fredrik/src/flint/src/acb.h:21,
                 from /home/fredrik/src/flint/src/acb/dot_siui.c:12:
/home/fredrik/src/flint/src/arb.h:416:6: note: by argument 6 of type ‘arb_srcptr’ {aka ‘const arb_struct *’} to ‘arb_dot’ declared here
  416 | void arb_dot(arb_t res, const arb_t initial, int subtract,
      |      ^~~~~~~


/home/fredrik/src/flint/src/acb/dot_ui.c: In function ‘acb_dot_ui’:
/home/fredrik/src/flint/src/acb/dot_ui.c:70:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
   70 |     arb_dot(((arb_ptr) res) + 0, (initial == NULL) ? NULL : ((arb_srcptr) initial) + 0, subtract, ((arb_srcptr) x) + 0, 2 * xstep, t, 1, len, prec);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/fredrik/src/flint/src/acb.h:21,
                 from /home/fredrik/src/flint/src/acb/dot_ui.c:12:
/home/fredrik/src/flint/src/arb.h:416:6: note: by argument 6 of type ‘arb_srcptr’ {aka ‘const arb_struct *’} to ‘arb_dot’ declared here
  416 | void arb_dot(arb_t res, const arb_t initial, int subtract,
      |      ^~~~~~~


/home/fredrik/src/flint/src/acb/dot_uiui.c: In function ‘acb_dot_uiui’:
/home/fredrik/src/flint/src/acb/dot_uiui.c:101:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
  101 |     arb_dot(((arb_ptr) res) + 0, (initial == NULL) ? NULL : ((arb_srcptr) initial) + 0, subtract, ((arb_srcptr) x) + 0, 2 * xstep, t, 1, len, prec);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/fredrik/src/flint/src/acb.h:21,
                 from /home/fredrik/src/flint/src/acb/dot_uiui.c:12:
/home/fredrik/src/flint/src/arb.h:416:6: note: by argument 6 of type ‘arb_srcptr’ {aka ‘const arb_struct *’} to ‘arb_dot’ declared here
  416 | void arb_dot(arb_t res, const arb_t initial, int subtract,
      |      ^~~~~~~

```

**Comment by fredrik-johansson** [comment]
I believe we can use

```
#pragma GCC diagnostic ignored "-Wuninitialized"
```

or rather we should add a macro for this as we've done for some other warnings.
